new-frontiers-14/frontier-station-14
# RPED Fixes, Minor Cleanup

## About the PR



Minor change to RPED replacement algo.  Removal of items from lists is now done by index instead of by value comparison.

Items have their status updated - elements that started in the RPED will not be reinserted into the RPED.  Elements that were removed from the machine _will_ be inserted into the RPED, if there's space.  They will be dropped on the ground if they do not fit.

## Why / Balance


Stack removal was buggy.  Occasionally, insertion would consume _all of the stacks of a type in your inventory_.  Easy way of replicating: spawn an RPED, a thruster upgrade kit, insert caps into RPED, upgrade a thruster.  Poof, no more super caps.

## How to test


1. Spawn RPED, thruster upgrade kit, thruster.
2. Load RPED with thruster upgrade kit.
3. Upgrade thruster.  RPED should have 8 capacitors left in it (note: currently in two stacks), the thruster should be upgraded with super capacitors.
4. Fill RPED with super capacitors, add six more capacitors.
5. Spawn another thruster.
6. Upgrade thruster.
7. 4 capacitors (separate stacks - different issue) should pop up on top of the thruster.  The RPED should have 4 super capacitors missing, and the thruster should be upgraded with super capacitors.

## Media



## Breaking changes


**Changelog**



:cl:
- fix: The RPED no longer consumes unused components when upgrading a machine.
